js/clientJS/imgUploadToDB.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * This file contains the functions to upload an image to the database
+ * Manly used to upload profile pictures
+ */
+
+/**
+ * Converts a file to a base64 string
+ * @param file js file object (from file input for example)
+ * @returns {Promise<unknown>} a Promise that resolves to the base64 string
+ */
+function convertBase64(file) {
+    return new Promise((resolve, reject) => {
+        const fileReader = new FileReader();
+        fileReader.readAsDataURL(file);
+
+        fileReader.onload = () => {
+            resolve(fileReader.result);
+        };
+
+        fileReader.onerror = (error) => {
+            reject(error);
+        };
+    });
+}
+
+/**
+ * Updates the profile picture of a user
+ * Only files with a size smaller than 10 MB are allowed
+ * @param event the event change from the file input
+ * @param userId the id of the user
+ * @returns {Promise<unknown>} Promise that resolves is the profile picture was updated successfully
+ */
```

Comment:
It might be a good idea to add an automatic resizing of the input file. Currently if you upload a large profile picture it gets saved with that resolution which is unecessary

Reply:
Thats true; Thumbnailing is another feature we should imlpement. Atm the picture gets saved full sized (max 10MB) thats because this picture might get used in other use cases in which reesolution matters.